core/objectstore/objectstore.go

```diff
@@ -97,11 +97,11 @@ type ReadObjectStore interface {
 // WriteObjectStore represents an object store that can only be written to.
 type WriteObjectStore interface {
 	// Put stores data from reader at path, namespaced to the model.
-	Put(ctx context.Context, path string, r io.Reader, size int64) error
+	Put(ctx context.Context, path string, r io.Reader, size int64) (UUID, error)
 
 	// Put stores data from reader at path, namespaced to the model.
 	// It also ensures the stored data has the correct hash.
-	PutAndCheckHash(ctx context.Context, path string, r io.Reader, size int64, hash string) error
+	PutAndCheckHash(ctx context.Context, path string, r io.Reader, size int64, hash string) (UUID, error)
 
 	// Remove removes data at path, namespaced to the model.
 	Remove(ctx context.Context, path string) error
```

Comment:
That's a bigger change than I'm willing to push through. It would require updating all the call sites of all object store access. We can do it later once we're further down the line.

Essentially it's a side quest I'm unwilling to follow.

You'll get RI from it, that is all.